javascript/new-src/src/js/app.js

```diff
@@ -14,520 +14,582 @@
-}
-])
-.controller('keypadUniverseCtrl', ['$scope', '$ola', '$routeParams', function ($scope, $ola, $routeParams) {
- "use strict";
- $ola.tabs('keypad', $routeParams.id);
- $scope.Universe = $routeParams.id;
-}])
-.controller('rdmUniverseCtrl', ['$scope', '$ola', '$routeParams', function ($scope, $ola, $routeParams) {
- "use strict";
- $ola.tabs('rdm', $routeParams.id);
- //get: /json/rdm/set_section_info?id={{id}}&uid={{uid}}&section={{section}}&hint={{hint}}&address={{address}}
- $scope.Universe = $routeParams.id;
-}])
-.controller('patchUniverseCtrl', ['$scope', '$ola', '$routeParams', function ($scope, $ola, $routeParams) {
- "use strict";
- $ola.tabs('patch', $routeParams.id);
- $scope.Universe = $routeParams.id;
-}])
-.controller('settingUniverseCtrl', ['$scope', '$ola', '$routeParams', function ($scope, $ola, $routeParams) {
- "use strict";
- $ola.tabs('settings', $routeParams.id);
- //post: /modify_universe
- //id:{{id}}
- //name:{{name}}
- //merge_mode:(LTP|HTP)
- //add_ports:{{port}}
- //remove_ports:{{port}}
- //{{port}}_priority_value:100
- //{{port}}_priority_mode:(inherit|static)
- //modify_ports:{{port}}
- $scope.Universe = $routeParams.id;
- $scope.Info = {};
- $scope.PortsId = {};
- $scope.ActivePorts = {};
- $scope.Remove = [];
- $scope.Data = {};
- $ola.get.PortsId($routeParams.id).then(function (data) {
-  $scope.PortsId = data;
- });
- $ola.get.UniverseInfo($routeParams.id).then(function (data) {
-  $scope.Info = data;
-  $scope.ActivePorts = data.output_ports.concat(data.input_ports);
-  for (var i = 0; i < $scope.ActivePorts.length; i++) {
-   $scope.Remove[i] = "";
-  }
- });
-}])
-.controller('pluginInfoCtrl', ['$scope', '$routeParams', '$ola', '$window',
-function ($scope, $routeParams, $ola, $window) {
- "use strict";
- $ola.get.InfoPlugin($routeParams.id).then(function (data) {
-  $scope.active = data.active;
-  $scope.enabled = data.enabled;
-  $scope.name = data.name;
-  var description = document.getElementById('description');
-  description.textContent = data.description;
-  description.innerHTML = description.innerHTML.replace(/\\n/g, '<br />');
-  $window.console.log(data);
- });
- $scope.stateColor = function (val) {
-  if (val) {
-   return {
-    'background-color': 'green'
+   var dmxGet = $interval(function () {
+    $ola.get.Dmx($scope.Universe).then(function (data) {
+     for (var i = 0; i < data.dmx.length; i++) {
+      $scope.get[i] = data.dmx[i];
+     }
+     $scope.send = true;
+    });
+   }, 1000);
+   $scope.getColor = function (i) {
+    if (i > 140) {
+     return 'black';
+    } else {
+     return 'white';
+    }
    };
-  } else {
-   return {
-    'background-color': 'red'
+   $scope.ceil = function (i) {
+    return $window.Math.ceil(i);
    };
+   $scope.change = function () {
+    $ola.post.Dmx($scope.Universe, $scope.get);
+   };
+   $scope.page = function (d) {
+    if (d === 1) {
+     if (($scope.offset + 1) !==
+      $window.Math.ceil(OLA.MAX_CHANNEL_NUMBER / $scope.limit)) {
```

Comment:
Okay, you could also wrap the divide across two lines, or move everything after ceil( onto a further indented line.